zproject/prod_settings_template.py

```diff
@@ -343,39 +343,37 @@
-## on it (`/complete/example/` if the name is "example") and the names of the settings you'll configure
-## are generated based on it.
-# SOCIAL_AUTH_GENERIC_OIDC_NAME = "example"
-##
-## (2) Adjust the names of the settings below according to the name you configured
-## in SOCIAL_AUTH_GENERIC_OIDC_NAME.
-## E.g. if the name is "example", the setting names will be as below.
-## If the name is changed to "anotherexample",
-## SOCIAL_AUTH_EXAMPLE_DISPLAY_NAME will have to be changed to SOCIAL_AUTH_ANOTHEREXAMPLE_DISPLAY_NAME etc.
-
-## The base path to the provider's OIDC API. Zulip fetches the IdP's configuration from the
-## discovery endpoint, whose URL is assumed to be the URL below with "/.well-known/openid-configuration"
-## appended at the end.
-SOCIAL_AUTH_EXAMPLE_OIDC_URL = "https://example.com/api/openid"
-## The display name, which will appear on the "Log in with <display name>" buttons.
-SOCIAL_AUTH_EXAMPLE_DISPLAY_NAME = "Example"
-## Optional: URL to the icon that should be displayed on the "Log in with ..." button.
-SOCIAL_AUTH_EXAMPLE_DISPLAY_ICON = None
 
+from .config import get_secret
+
+## Note: The OIDC IdP is configured in a dictionary format
+## to allow future support of multiple IdPs, but for now
+## only one is allowed.
+SOCIAL_AUTH_OIDC_ENABLED_IDPS = {
+    ## You can change the idp_name below to a preferred name, but it
+    ## doesn't affect anything user-facing.
+    "idp_name": {
+        ## The base path to the provider's OIDC API. Zulip fetches the IdP's configuration from the
+        ## discovery endpoint, whose URL is assumed to be the URL below with "/.well-known/openid-configuration"
+        ## appended at the end.
+        "oidc_url": "https://example.com/api/openid",
+        ## The display name, which will appear on the "Log in with <display name>" buttons.
+        "display_name": "Example",
+        ## Optional: URL to the icon that should be displayed on the "Log in with ..." button.
+        "display_icon": None,
+        ## Now configure the client id and secret.
+        ## Specify the client id below and put the client secret in
+        ## zulip-secrets.conf under a name you choose and use that name
+        ## as the argument to the get_secret call below.
+        "client_id": "<your client id>",
+        "secret": get_secret("social_auth_oidc_secret"),
+    }
+}
 ## You can set the setting below to True to make Zulip "trust" the Full Name
 ## provided by the IdP at the userinfo endpoint. This will allow skipping
 ## the registration form in the sign up process (unless TERMS_OF_SERVICE need to be accepted)
 ## the user won't be asked for their Full Name anymore.
-# SOCIAL_AUTH_EXAMPLE_FULL_NAME_VALIDATED = True
-
-## Now configure the client id and secret.
-## Specify the client id below and put the client secret in
-## zulip-secrets.conf as `social_auth_example_secret`.
-# SOCIAL_AUTH_EXAMPLE_KEY = <your client ID>
+SOCIAL_AUTH_OIDC_FULL_NAME_VALIDATED = True
```

Comment:
This probably belongs in the `SOCIAL_AUTH_OIDC_ENABLED_IDPS` dict, right?

Reply:
Hmm, it's a separate setting on purpose because currently the `full_name_validated` thing is a class attribute of the backend - we'd need a completely different way of plumbing the information into the social pipeline to support varying it with IdP.